### Introduction
#### Mission Statement: Rooted in Rights tells authentic, accessible stories to challenge stigma and redefine narratives around disability, mental health, and chronic illness. As part of Disability Rights Washington, our Seattle-based team of disabled video producers, editors, and digital organizers partner with both local coalitions and national advocacy campaigns to fight for concrete changes for our community.  
#### Rooted in Rights is a project of Disability Rights Washington. https://www.disabilityrightswa.org  Anna Letitia Zivarts (206) 718-8432 (206.324.1521 x242), Program Director for RootedinRights.org. https://rootedinrights.org/   
### Business Understanding  
#### This project will attempt to answer the following questions (numbered from the initial meeting):  
##### 1) Number of households in Washington State where there are more people than vehicles, as a measure to show people who don't have consistent access to a car for all of their transit needs.   
##### 4) Number or percentage of individuals in Washington State who have a disability that prevents them from driving (this would include elders). I would love to see this data broken down by county for our state.  
### Analytic Approach  
#### (No machine-learning algorithms required yet)  
### Data Requirements  
#### Answer questions 1 and 4  
#### We're going to need relevent and applicable structured  or un-structured data in electronic form, like a database or flatfile, obtainable via online channels, that can be downloaded. The national survey of 2017 when reduced to only Washington resident respondents, does not appear to contain a sufficient sample size to answer the subject questions. 

### Data Collection

In [2]:
# Import libraries
import pandas as pd
import numpy as np
import xlrd

In [3]:
# Use settings
# pd.set_option('display.width', None)
# print(pd.options.display.max_rows)
# pd.set_option('display.max_rows', 40)

#### We're going to use the perpub.csv file only for these two questions

In [4]:
# Access local files
# path_hhpub = './data/rootedinrights/hhpub.csv'
path_perpub = './data/rootedinrights/perpub.csv'
# path_trippub = './data/rootedinrights/trippub.csv'
# path_vehpub = './data/rootedinrights/vehpub.csv'
# path_nhts = './data/rootedinrights/NHTS_2009_transfer_WA.txt'
# path_latch = './data/rootedinrights/latch_2017-b.csv'

In [5]:
import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

In [6]:
# The code was removed by Watson Studio for sharing.

In [7]:
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )
    
# If you are reading an Excel file into a pandas DataFrame, replace `read_csv` by `read_excel` in the next statement.
df_perpub = pd.read_csv(body)
df_perpub.head()

HOUSEID  PERSONID  R_AGE  EDUC  R_HISP  R_RELAT  R_SEX  R_RACE  PRMACT  \
0  30000007         1     67     3       2        1      2       2       6   
1  30000007         2     66     3       2        2      1       2       1   
2  30000007         3     28     2       2        3      2       2       5   
3  30000008         1     55     5       2        1      1       1       1   
4  30000008         2     49     4       2        2      2       1       1   

   PAYPROF  ...  SMPLSRCE    WTPERFIN  HBHUR  HTHTNRNT  HTPPOPDN  HTRESDN  \
0        2  ...         2  206.690153      T        50      1500      750   
1       -1  ...         2  197.075742      T        50      1500      750   
2        2  ...         2  219.514210      T        50      1500      750   
3       -1  ...         2   63.185911      R         5       300      300   
4       -1  ...         2   58.665911      R         5       300      300   

   HTEEMPDN  HBHTNRNT  HBPPOPDN  HBRESDN  
0       750        20       750      300  
1       750        20       750      300  
2       750        20       750      300  
3       150         5       300      300  
4       150         5       300      300  

[5 rows x 121 columns]

In [8]:
# open a text file
# with open(path_nhts) as file_nhts:
#     # for line in file_nhts:
#     #     print(line)
#     print(file_nhts)
# df_nhts = pd.read_csv(path_nhts, sep=None, engine='python', dtype={'geoid':np.int64, 'Cluster':np.float64})

### Data Understanding  
#### Check the dataframe information

In [9]:
# print('df_hhpub', df_hhpub.shape,'\n', df_hhpub.head(10), '\n')         # (129696, 58)
# print(df_hhpub.info(memory_usage=True, null_counts=True))               # dtypes: float64(1), int64(54), object(3)

# We can check to see how many null objects we have in the dataset columns as follows:
# arenulls = df_hhpub.isnull().any() # or use .sum()
# print(arenulls)       # There are none
# for c in range(len(df_hhpub.columns)):
#     c_column = df_hhpub.columns[c]
#     print(df_hhpub[c_column].isnull().value_counts(dropna=False))       # there are none

# Print the feature names in one column for easy lookup in the Codebook.pdf
# for f in sorted(df_hhpub.columns.tolist()):
#     print(f)

# @@@@@
# print('df_perpub', df_perpub.shape,'\n', df_perpub.head(10), '\n')      # (264234, 121)
df_perpub.info(memory_usage=True, null_counts=True)                    # dtypes: float64(4), int64(113), object(4)
# for c in range(len(df_perpub.columns)):
#     c_column = df_perpub.columns[c]
#     print(df_perpub[c_column].isnull().value_counts(dropna=False))       # there are none
# for f in sorted(df_perpub.columns.tolist()):
#     print(f)

# # @@@@@
# print('df_trippub', df_trippub.shape,'\n', df_trippub.head(10), '\n')      # (923572, 105)
# print(df_trippub.info(memory_usage=True, null_counts=True))                # dtypes: float64(4), int64(97), object(4)
# for c in range(len(df_trippub.columns)):
#     c_column = df_trippub.columns[c]
#     print(df_trippub[c_column].isnull().value_counts(dropna=False))        # there are none
# for f in sorted(df_trippub.columns.tolist()):
#     print(f)
# @@@@@
# print('df_vehpub', df_vehpub.shape,'\n', df_vehpub.head(10), '\n')          # (256115, 54)
# print(df_vehpub.info(memory_usage=True, null_counts=True))                # dtypes: float64(2), int64(47), object(5)
# for c in range(len(df_vehpub.columns)):
#     c_column = df_vehpub.columns[c]
#     print(df_vehpub[c_column].isnull().value_counts(dropna=False))        # there are none
# for f in sorted(df_vehpub.columns.tolist()):
#     print(f)

# @@@@@
# print('df_nhts', df_nhts.shape,'\n', df_nhts.head(20), '\n')          # (1458, 116)     All the feature values after 'geoid' are NaNs. EDIT: determined that the table authors had left the 'cells' blank
# print(df_nhts.info(memory_usage=True, null_counts=True), '\n')                # dtypes: float64(115), int64(1)
# for c in range(len(df_nhts.columns)):
#     c_column = df_nhts.columns[c]
#     print(df_nhts[c_column].isnull().value_counts(dropna=True))        # the first feature 'geoid' is all int64. Initially, all subsequent coulumns are int64 @ 1446 rows with 46 NaNs in the earliest rows 0 ~ 11
# nhts_cols = df_nhts.columns.tolist()
# for f in nhts_cols:
#     print(f)
# print(nhts_cols)

# @@@@@
# print('df_latch', df_latch.shape,'\n', df_latch.head(20), '\n')          # (73056, 128)
# print(df_latch.info(memory_usage=True, null_counts=True), '\n')          # dtypes: float64(100), int64(28)
# for c in range(len(df_latch.columns)):
#     c_column = df_latch.columns[c]
#     print(df_latch[c_column].isnull().value_counts(dropna=True))        # the table has lots of NaNs

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 264234 entries, 0 to 264233
Columns: 121 entries, HOUSEID to HBRESDN
dtypes: float64(4), int64(113), object(4)
memory usage: 243.9+ MB


#### It appears that perpub.csv has 264,234 observations with 121 columns (features), and that there are 4 - float64, 113 - int64, and 4 - object dtypes 

### Data Preparation

In [10]:
# Handle NaNs
# df_nhts.dropna(inplace = True)
# df_nhts.fillna(0, inplace=True)
# df_nhts = df_nhts.astype({'geoid':int,'Cluster':int,'urban_group':int})
# # Check
# print('df_nhts', df_nhts.shape,'\n', df_nhts.head(20), '\n')
# print('pct_veh_2_more' in df_nhts.columns.values)      # feature present in table: 'tot_pop'>>False, 'hh_cnt'>>False, 'hh_veh_1'>>False, 'pct_veh_1':False , 'hh_veh_2_more':False, 'pct_veh_2_more':False

# @@@@@
# print('CONDRIVE' in df_latch.columns.values)      # feature present in table: 'tot_pop'>>True, 'hh_cnt'>>True, 'hh_veh_1'>>True, 'pct_veh_1':True, 'hh_veh_2_more':True, 'pct_veh_2_more':True, 'SAMEPLC':False, 'CONDRIVE':False
# Handle NaNs
# df_latch.fillna(0, inplace=True)
# Adjust dtypes
# df_latch = df_latch.astype({'':int,'':int,'':int})

# Cluster 6
# df_latch = df_latch[['geocode','Cluster','tot_pop','hh_cnt','hh_veh_1','hh_veh_2_more','pct_veh_1','pct_veh_2_more']]     # (2052, 8)
# Filter the table for a Cluster = 6 orientation. According to USDOT, Bureau of Transportation Statistics, latch_2017-b.csv, census regions and divisions are combined to form a sample of sufficient size
# filter = df_latch['Cluster'] == 6
# df_latch.where(filter, inplace=True)      # some features dtype will change when using this method
# df_latch = df_latch[filter]

# State FIP codes- 2:Alaska, 6:California, 15:Hawaii, 41:Oregon, 53:Washington
# Extract only the observations which have a FIP = 53 in the feature 'geocode'
# cntr = 0
# for g in range(len(df_latch)):
#     if str(df_latch['geocode'][g])[:2] == '53':
#         cntr += 1
#         print(cntr, '\t', str(df_latch['geocode'][g])[:2])
# filter = (df_latch['geocode'] >= 53000000000) & (df_latch['geocode'] < 54000000000)
# df_latch = df_latch[filter]
# df_latch.reset_index(inplace=True)

# Drop the NaN rows
# df_latch.dropna(inplace = True, how='all')
# Check
# print('df_latch', df_latch.shape,'\n', df_latch, '\n')

#### We're going to use several of the following columns (features) for this analysis. We have the feature name in uppercase followed by the data description, and the refering page in the table descriptive guide (1)

In [11]:
# @@@@@
# R_AGE: Age. Page 40 for codes (reply of the respondents age)
# DRVRCNT: number of drivers in household. Page 25 for codes
# HHVEHCNT: Count of household vehicles. Page 58 for codes
# NUMADLT: Count of adult household members at least 18 years old. Page 75/6 for codes
# HHSIZE: Count of household members. Counts from 1 ~ 13. Page 28 for codes
# CONDRIVE: Medical Condition Results in Giving up Driving. Page 24 for codes. "Medical condition results in giving-up driving" 1 = yes
# MEDCOND: Medical condition. '01' = Yes, '02' = No. Page 36 for codes
# CONRIDE: Medical Condition Results in Asking Others for Rides. Page 24 for codes
# CONDSPEC: Medical Condition Results in Using Special Transportation Services. Page 24/5 for codes
# SAMEPLC: Reason for No Trips. Refer to page 41 for codes
# CONDTRAV: Medical condition results in reduced day-to-day travel.  'CONDTRAV' = 01/yes. Page 25 for codes.
# ALT_23: Alternative Mode of Transportation: Passenger to Friend/Family Member or Rental Car. Page 17 for codes
# ALT_16: Alternative mode of transportation: Public Transportation or Taxi. Page 17 for codes
# HHSTATE: household state. Page 28/9 for codes
# HHSTFIPS: State FIPS for household address. 53 = Washington state. Page 58 for codes

# Not used
# CENSUS_D: 2010 Census division classification for the respondent's home address. Page 53 for codes
# CENSUS_R: Census region classification for home address. Page 53 for codes
# HBHUR: Urban / Rural indicator - Block group. Page 54 for codes

# At a confidence level of 95% and a confidence interval (margin of error) of +/-3, the smallest sample size is 1067 for  a population of 7,500,000. This assumes, of course, that the sample is genuinely random.
# Source calculator: https://www.surveysystem.com/sscalc.htm

#### Extract the Washington state respondents from the national survey

In [12]:
# Use settings
pd.set_option('display.width', None)
pd.set_option('display.max_rows', 10)

In [13]:
condition = df_perpub['HHSTATE'] == 'WA'
df_perpub_WA = df_perpub[condition]
df_perpub_WA = df_perpub_WA[['R_AGE','DRVRCNT','HHVEHCNT','NUMADLT','HHSIZE','CONDRIVE', 'MEDCOND', 'CONDRIDE','CONDSPEC','SAMEPLC','CONDTRAV','ALT_23','ALT_16','HHSTATE','HHSTFIPS']]
df_perpub_WA.reset_index(inplace=True)
df_perpub_WA.drop(['index'], axis=1, inplace=True)
# Check
print('df_perpub_WA', df_perpub_WA.shape,'\n', df_perpub_WA, '\n')      # (1326, # of columns)

df_perpub_WA (1326, 15) 
       R_AGE  DRVRCNT  HHVEHCNT  NUMADLT  HHSIZE  CONDRIVE  MEDCOND  CONDRIDE  \
0        92        3         3        3       6         2        2         2   
1        42        3         3        3       6        -1        2        -1   
2        40        3         3        3       6        -1        2        -1   
3        15        3         3        3       6        -1        2        -1   
4         7        3         3        3       6        -1        2        -1   
...     ...      ...       ...      ...     ...       ...      ...       ...   
1321     62        3         3        3       3        -1        2        -1   
1322     63        3         3        3       3        -1        2        -1   
1323     32        3         3        3       3        -1        2        -1   
1324     60        1         1        1       1        -1        2        -1   
1325     59        1         2        1       1        -1        2        -1   

      CONDSPE

#### We now see that the sample segment from Washington state is only 1326 respondents described by 15 columns (features)

#### Let's extract some count and percentile information about the national-level survey

In [14]:
# @@@@
people_count_perpub = len(df_perpub)
print(f"From the national-level survey sample of ({people_count_perpub}):\n")
# Determine the segment count and percentage of national-level respondents who answered "reason for no trips" as "04 - Disabled or home-bound"

disabled = 0
for pc in df_perpub['SAMEPLC']:
    # '04' means the respondent is disabled or is homebound
    if pc == 4:
        disabled += 1

print(f"{disabled} respondents ({disabled/people_count_perpub*100:.2f}%) say that they are disabled or are home-bound")

# Determine the segment count and percentage of people who have a medical condition
haveMedCond_natl_count = 0
for h in df_perpub['MEDCOND']:
    # Check value
    # print(h)
    # 1 means the respondent has a medical condtion
    if h == 1:
        haveMedCond_natl_count += 1

print(f"{haveMedCond_natl_count} respondents ({haveMedCond_natl_count/people_count_perpub*100:.2f}%) stated that they have a medical condition")

# Determine the segment count and percentage of people who have given-up driving due to medical condition
nodrive_natl_count = 0
# drive_count = 0
for p in df_perpub['CONDRIVE']:
    # Check value
    # print(p)
    # 1 means 'have given up driving'
    if p == 1:
        nodrive_natl_count += 1

print(f"{nodrive_natl_count} respondents ({nodrive_natl_count/people_count_perpub*100:.2f}%) stated that they've given-up driving due to their medical condition.")

# Determine the segment count and percentage of people who have "Medical Condition Results in Reduced Day-to-Day Travel"
reduced_travel_count = 0
for r in df_perpub['CONDTRAV']:
    # Check value
    # 1 means 'yes,' medical condition reduces day-to-day travel
    if r == 1:
        reduced_travel_count += 1
        
print(f"{reduced_travel_count} respondents ({reduced_travel_count/people_count_perpub*100:.2f}%) stated that their medical condition reduces day-to-day travel.\n")
print(f"The sum of travel-related medical condition/disabilities nationwide is: {(disabled + haveMedCond_natl_count + nodrive_natl_count + reduced_travel_count)/people_count_perpub*100:.2f}%")

From the national-level survey sample of (264234):

3913 respondents (1.48%) say that they are disabled or are home-bound
25181 respondents (9.53%) stated that they have a medical condition
6806 respondents (2.58%) stated that they've given-up driving due to their medical condition.
21028 respondents (7.96%) stated that their medical condition reduces day-to-day travel.

The sum of travel-related medical condition/disabilities nationwide is: 21.54%


#### Let's compare the summary information above to the Washington segment sample below.  

In [15]:
# @@@@
people_count = len(df_perpub_WA)
# print(people_count)
print(f"From the Washington state survey segment of ({people_count}):\n")

# Determine the segment count and percentage of people in WA who answered "reason for no trips" as "04 - Disabled or home-bound"
homebound = 0
for hb in df_perpub_WA['SAMEPLC']:
    # '04' means the respondent is disabled or is homebound
    if hb == 4:
        homebound += 1

print(f"{homebound} ({homebound/people_count*100:.2f}%) of respondents say that they are disabled or are home-bound [SAMEPLC = 4]")

# Determine the segment count and percentage of people in WA who have a medical condition
haveMedCond_count = 0
for h in df_perpub_WA['MEDCOND']:
    # Check value
    # print(h)
    # 1 means the respondent has a medical condtion
    if h == 1:
        haveMedCond_count += 1

print(f"{haveMedCond_count} respondents ({haveMedCond_count/people_count*100:.2f}%) stated that they have a medical condition [MEDCOND = 1]")

# Determine the segment count and percentage of people in WA who have given-up driving due to medical condition
nodrive_count = 0
# drive_count = 0
for p in df_perpub_WA['CONDRIVE']:
    # Check value
    # print(p)
    # 1 means 'have given up driving'
    if p == 1:
        nodrive_count += 1

print(f"{nodrive_count} respondents ({nodrive_count/people_count*100:.2f}%) stated that they've given-up driving due to their medical condition [CONDRIVE = 1]")

# Determine the segment count and percentage of people in WA who have "Medical Condition Results in Reduced Day-to-Day Travel"
reduced_travel_WA_count = 0
for r in df_perpub_WA['CONDTRAV']:
    # Check value
    # 1 means 'yes,' medical condition reduces day-to-day travel
    if r == 1:
        reduced_travel_WA_count += 1
        
print(f"{reduced_travel_WA_count} respondents ({reduced_travel_WA_count/people_count*100:.2f}%) stated that their medical condition reduces day-to-day travel [CONDTRAV = 1]\n")
print(f"The sum of travel-related medical condition/disabilities in Washington state is: {(homebound + haveMedCond_count + nodrive_count + reduced_travel_WA_count)/people_count*100:.2f}%")

From the Washington state survey segment of (1326):

13 (0.98%) of respondents say that they are disabled or are home-bound [SAMEPLC = 4]
113 respondents (8.52%) stated that they have a medical condition [MEDCOND = 1]
30 respondents (2.26%) stated that they've given-up driving due to their medical condition [CONDRIVE = 1]
96 respondents (7.24%) stated that their medical condition reduces day-to-day travel [CONDTRAV = 1]

The sum of travel-related medical condition/disabilities in Washington state is: 19.00%


#### The Travel Patterns of American Adults with Disabilities Report (4) stated:  
> "...people with disabilities simply travel less often:
    > * Reducing day-to-day travel (70.6 percent)
    > * Giving up driving (21.6 percent)
    > * Using public transit less often (14.4 percent)  
    
#### Using the national-level data we see that 6808 (2.58%) respondents have given-up driving due to their medical condition. The Washington segment shows that 30 (2.26%) of respondents have reported the same; these percentiles are roughly similar. Given these percentiles it would appear that the Washington segment is not too different from the national segment. However, between the two reports there appears to be a large difference in the "given-up driving" category. Ms. Stephanie Lawrence (Bureau of Transportation Statistics) reponded to this very question:

#### S. Lawrence,  (personal communication, December 30, 2019) advises a cautionary note concerning the BTS Travel Patterns of American Adults with Disabilities Report:
> "One thing I would caution is that State-level estimates of travel by people with disabilities are unreliable due to small sample sizes and large standard errors from the NHTS. Using the Census region (CENSUS_R) will yield more usable estimates, although I suspect they won’t differ much from national estimates.  
The “21.6 percent” of people who give up driving is “21.6 percent of people age 18–64 who have a travel-limiting disability,” which likely explains the discrepancy. In addition, the NHTS asks the CONDTRAV question to everyone 80 or older—whether or not they report having a disability—as well as others who report having a disability. This fact isn’t apparent from the codebook, but appears in the questionnaire at https://nhts.ornl.gov/assets/2016/NHTS_Retrieval_Instrument_20180228.pdf "  

#### Because the usage of column features such as HHSTATE, CENSUS_R, HHSTFIPS allows us to separate Washington-specific information from the national counts and draw us closer toward our answer, they still do not provide a firm basis for inference to our state population. Perhaps then, we should approach the usage of these percentiles with a grain-of-salt until proper data is available. That is to say, Washington-specific survey information, since our ability to produce an appropriate geographic sample size is limited.

#### Nonetheless, lets learn a bit more about the household situation of the people who have given-up driving because of their medical condition based on the data we have.

In [16]:
# Specifically, the number of drivers in the household
condition1 = df_perpub_WA['CONDRIVE'] == 1
df_condrive = df_perpub_WA[condition1]
df_condrive.index = pd.RangeIndex(len(df_condrive.index))
# print('df_condrive', df_condrive.shape,'\n', df_condrive, '\n')      # (30, # of columns)

nodrive_hh_adult_counttypes = df_condrive['NUMADLT'].unique()
# print(nodrive_hh_adult_counttypes)
adult_count_1 = 0
adult_count_2 = 0
adult_count_3 = 0

list_size = df_condrive['HHSIZE'].unique()
# print(list_size)
nodrive_hhsize = dict((el,0) for el in list_size)
adult_1_hhsize = dict((el,0) for el in list_size)
adult_2_hhsize = dict((el,0) for el in list_size)
adult_3_hhsize = dict((el,0) for el in list_size)
# print(nodrive_hhsize)

# convert this loop to use index and range instead of value
for idx in range(len(df_condrive)):
    # print(adult)
    if 1 == df_condrive['NUMADLT'][idx]:
        adult_count_1 += 1
        # Update 'adult_1_hhsize' dictionary for the HHSIZE count
        if df_condrive['HHSIZE'][idx] in list_size:
            adult_1_hhsize[df_condrive['HHSIZE'][idx]] += 1
    elif 2 == df_condrive['NUMADLT'][idx]:
        adult_count_2 += 1
        # Update 'adult_2_hhsize' dictionary for the HHSIZE count
        if df_condrive['HHSIZE'][idx] in list_size:
            adult_2_hhsize[df_condrive['HHSIZE'][idx]] += 1
    elif 3 == df_condrive['NUMADLT'][idx]:
        adult_count_3 += 1
        # Update 'adult_3_hhsize' dictionary for the HHSIZE count
        if df_condrive['HHSIZE'][idx] in list_size:
            adult_3_hhsize[df_condrive['HHSIZE'][idx]] += 1

# Delete the zero-value pairs in the dictionaries
adult_1_hhsize = {k:v for k,v in adult_1_hhsize.items() if v}       # {1: 8}
adult_2_hhsize = {k:v for k,v in adult_2_hhsize.items() if v}       # {2: 13, 6: 1, 4: 1}
adult_3_hhsize = {k:v for k,v in adult_3_hhsize.items() if v}       # {3: 6, 5: 1}      ({/*100:.2f}%)

# print('\n', adult_1_hhsize, '\n', adult_2_hhsize, '\n', adult_3_hhsize, '\n')

print(f"This group of 30 respondents had the following count of adults and household size distributions (number of people total in household):\n\
    1 adult: {adult_count_1} ({adult_count_1/nodrive_count*100:.2f}%) instances. {adult_1_hhsize[1]} ({adult_1_hhsize[1]/adult_count_1*100:.2f}%) households with {[key for key in adult_1_hhsize.keys()][0]} person\n\
\
    2 adults: {adult_count_2} ({adult_count_2/nodrive_count*100:.2f}%) instances. {adult_2_hhsize[2]} ({adult_2_hhsize[2]/adult_count_2*100:.2f}%) households with {[key for key in adult_2_hhsize.keys()][0]} people, {adult_2_hhsize[4]} ({adult_2_hhsize[4]/adult_count_2*100:.2f}%) household with {[key for key in adult_2_hhsize.keys()][2]} people, {adult_2_hhsize[6]} ({adult_2_hhsize[4]/adult_count_2*100:.2f}%) household with {[key for key in adult_2_hhsize.keys()][1]} people\n\
\
    3 adults: {adult_count_3} ({adult_count_3/nodrive_count*100:.2f}%) instances. {adult_3_hhsize[3]} ({adult_3_hhsize[3]/adult_count_3*100:.2f}%) households with {[key for key in adult_3_hhsize.keys()][0]} people, {adult_3_hhsize[5]} ({adult_3_hhsize[5]/adult_count_3*100:.2f}%) household with {[key for key in adult_3_hhsize.keys()][1]} people")

This group of 30 respondents had the following count of adults and household size distributions (number of people total in household):
    1 adult: 8 (26.67%) instances. 8 (100.00%) households with 1 person
    2 adults: 15 (50.00%) instances. 13 (86.67%) households with 2 people, 1 (6.67%) household with 4 people, 1 (6.67%) household with 6 people
    3 adults: 7 (23.33%) instances. 6 (85.71%) households with 3 people, 1 (14.29%) household with 5 people


### References
1. U.S. Department of Transportation, Federal Highway Administration, 2017 National Household Travel Survey Users Guide: https://nhts.ornl.gov/assets/NHTS2017_UsersGuide_04232019_1.pdf  
2. 2017 National Household Travel Survey Codebook: https://nhts.ornl.gov/assets/codebook_v1.1.pdf  
3. National Household Travel Survey: https://nhts.ornl.gov  
4. U.S. Department of Transportation, Bureau of Transportation Statistics, Travel Patterns of American Adults with Disabilities. https://www.bts.gov/sites/bts.dot.gov/files/docs/explore-topics-and-geography/topics/passenger-travel/222466/travel-patterns-american-adults-disabilities-11-26-19.pdf